bucketlist/views.py

```diff
@@ -0,0 +1,223 @@
+from flask import jsonify, request, g
+from bucketlist import app, db
+from bucketlist.models import User, Bucketlist, BucketlistItem, ValidationError
+from bucketlist.auth import auth_token, verify_password, generate_auth_token
+
+
+@app.route("/auth/register", methods=["POST"])
+def new_user():
+    """
+    Creates a new user.
+    """
+    user = User()
+    try:
+        # validates user key/value inputs using a try-catch block
+        sanitized = user.import_data(request.json)
+        if sanitized == "Invalid":
+            return jsonify({"Message": "Username and Password are required"}), 400
+    except ValidationError as e:
+        return jsonify({"Message": str(e)}), 400
+
+    # check for duplicates before creating the new user
+    duplicate = User.query.filter_by(username=user.username).first()
```

Comment:
You could reduce these instead of checking if a user exists, put it in a try..except and look for Integrity error. It will reduce the number of ifs and checks